src/devices/Hcsr04/Hcsr04.cs

```diff
@@ -90,39 +92,39 @@ private bool TryGetDistance(out double result)
             // 100ms was chosen because max measurement time for this sensor is around 24ms for 400cm
             // additionally we need to account 60ms max delay.
             // Rounding this up to a 100 in case of a context switch.
-            long hangTicks = Environment.TickCount + 100;
+            long hangTicks = DateTime.Now.Ticks + 100 * TimeSpan.TicksPerMillisecond;
             _timer.Reset();
 
             // Measurements should be 60ms apart, in order to prevent trigger signal mixing with echo signal
             // ref https://components101.com/sites/default/files/component_datasheet/HCSR04%20Datasheet.pdf
-            while (Environment.TickCount - _lastMeasurment < 60)
+            while (DateTime.Now.Ticks - _lastMeasurment < 60 * TimeSpan.TicksPerMillisecond)
             {
-                Thread.Sleep(TimeSpan.FromMilliseconds(Environment.TickCount - _lastMeasurment));
+                Thread.Sleep(TimeSpan.FromTicks(DateTime.Now.Ticks - _lastMeasurment));
             }
 
+            _lastMeasurment = DateTime.Now.Ticks;
+
             // Trigger input for 10uS to start ranging
             _controller.Write(_trigger, PinValue.High);
-            Thread.Sleep(TimeSpan.FromMilliseconds(0.01));
+            DelayHelper.DelayMicroseconds(10, true);
```

Comment:
👍

Reply:
I have a commit delay :-) That was fixed as well! forgot to save the file originally!